### PR TITLE
[FLOC-2113] Support lack of storage backend configuration for Vagrant runner

### PR DESCRIPTION
**Pull request by itamarst**
Possibly a hacky solution and we should've just changed config somewhere, but seemed expedient.

### CODE REVIEW

**Comment by adamtheturtle** [comment]
Oh no, this has revealed that all the acceptance tests fail against Vagrant.

File: admin/acceptance.py [hunk 1]
```diff
@@ -621,8 +620,7 @@ def _provider_config_missing(self, provider):
             "{!r} config stanza.".format(provider)
         )
 
-    def _runner_VAGRANT(self, package_source,
-                        dataset_backend, provider_config):
+    def _runner_VAGRANT(self, package_source, provider_config):
         """
         :param PackageSource package_source: The source of omnibus packages.
         :param DatasetBackend dataset_backend: A ``DatasetBackend`` constant.
```
Comment (adamtheturtle) [inline admin/acceptance.py]: Remove this documentation.

File: admin/acceptance.py [hunk 2]
```diff
@@ -631,6 +629,16 @@ def _runner_VAGRANT(self, package_source,
...
         """
+        # By default Vagrant runner just uses ZFS, which is pre-configured
+        # in the Vagrant images, so additional configuration may not be
+        # available:
+        try:
+            dataset_backend = self.dataset_backend()
+            dataset_backend_config = self.dataset_backend_configuration()
+        except KeyError:
+            dataset_backend = DatasetBackend.zfs
+            dataset_backend_config = {}
```
Comment (adamtheturtle) [inline admin/acceptance.py]: I'm hoping that `"backend": "zfs"` is required here and that accounts for some part of the acceptance test failures.

Comment (adamtheturtle) [inline admin/acceptance.py]:
```python
    dataset_backend_configuration.update({
        u"backend": dataset_backend.name,
    })
```

seems not.

**Comment by adamtheturtle** [comment]
It is hard to tell if any test failures are caused by this. I can't see why they would be but it would be nice to know. If none of the test failures are caused by this I'd suggest merging and then creating follow-ups blocking FLOC-2008 for the Vagrant acceptance test failures. I'm struggling to search the 119 MB remote log file - maybe I should try Elasticsearch or similar.

**Comment by adamtheturtle** [comment]
Ah I've found an error:

```
Jun 06 21:35:42 node2 flocker-dataset-agent[4312]: {&#34;task_uuid&#34;: &#34;71bade93-99cd-499a-9b46-d8d1a794d0a2&#34;, &#34;error&#34;: true, &#34;timestamp&#34;: 1433626541.972202, &#34;message&#34;: &#34;Unhandled Error\nTraceback (most recent call last):\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/flocker/node/_deploy.py\&#34;, line 302, in run\n    RemoteVolumeManager(destination))\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/flocker/volume/service.py\&#34;, line 352, in handoff\n    changing_owner = pushing.addCallback(pushed)\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/twisted/internet/defer.py\&#34;, line 307, in addCallback\n    callbackKeywords=kw)\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/twisted/internet/defer.py\&#34;, line 296, in addCallbacks\n    self._runCallbacks()\n--- &lt;exception caught here&gt; ---\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/twisted/internet/defer.py\&#34;, line 578, in _runCallbacks\n    current.result = callback(current.result, *args, **kw)\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/flocker/volume/service.py\&#34;, line 350, in pushed\n    remote_uuid = destination.acquire(volume)\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/flocker/volume/_ipc.py\&#34;, line 144, in acquire\n    volume.name.to_bytes()]).decode(\&#34;ascii\&#34;)\n  File \&#34;/opt/flocker/lib/python2.7/site-packages/flocker/common/_ipc.py\&#34;, line 90, in get_output\n    raise IOError(\&#34;Bad exit\&#34;, remote_command, e.returncode, e.output)\nexceptions.IOError: (&#39;Bad exit&#39;, [&#39;flocker-volume&#39;, &#39;--config&#39;, &#39;/etc/flocker/volume.json&#39;, &#39;acquire&#39;, &#39;4e4328d8-2027-4fcd-9897-12e049aa9402&#39;, &#39;default.3c4b861f-e637-4a57-b272-1f3904c42e78&#39;], 1, &#39;&#39;)\n&#34;, &#34;message_type&#34;: &#34;twisted:log&#34;, &#34;task_level&#34;: [2, 5]}
```

**Comment by exarkun** [comment]
Simpler fix in #1522

**Pull request closed without merging**